File: src/ELibrary.Backend/UserApi/Domain/Dtos/Responses/UserAuthenticationResponse.cs
﻿namespace UserApi.Domain.Dtos.Responses
{
    public class UserAuthenticationResponse
    {
        public AuthToken? AuthToken { get; set; }
        public string Email { get; set; } = string.Empty;
        public List<string> Roles { get; set; } = [];
    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/auth/responses/userAuthenticationResponse.ts
import { AuthTokenResponse, mapAuthTokenResponseToAuthToken, UserAuth } from "../../../..";

export interface UserAuthenticationResponse {
    authToken: AuthTokenResponse;
    email: string;
    roles: string[];
}

export function mapUserAuthenticationResponseToUserAuthentication(response: UserAuthenticationResponse): UserAuth {
    return {
        isAuthenticated: true,
        authToken: mapAuthTokenResponseToAuthToken(response.authToken),
        email: response.email,
        roles: response.roles
    }
}